beekeeper-metadata-cleanup/src/main/java/com/expediagroup/beekeeper/metadata/cleanup/handler/MetadataHandler.java

```diff
@@ -16,16 +16,48 @@
 package com.expediagroup.beekeeper.metadata.cleanup.handler;
 
 import java.time.LocalDateTime;
+import java.util.List;
 
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
 
 import com.expediagroup.beekeeper.core.model.HousekeepingMetadata;
+import com.expediagroup.beekeeper.metadata.cleanup.cleaner.MetadataCleanup;
 
-public interface MetadataHandler {
+public class MetadataHandler {
 
-  Page<HousekeepingMetadata> findRecordsToClean(LocalDateTime instant, Pageable pageable);
+  private final MetadataCleanup metadataCleanup;
 
-  Pageable processPage(Pageable pageable, LocalDateTime instant, Page<HousekeepingMetadata> page,
-      boolean dryRunEnabled);
+  public MetadataHandler(MetadataCleanup metadataCleanup) {
+    this.metadataCleanup = metadataCleanup;
+  }
+
+  /**
+   * Processes a pageable HouseKeepingMetadata page.
+   *
+   * @param pageable Pageable to iterate through for dryRun
+   * @param instant Instant at which the cleanup is taking place
+   * @param page Page to get content from
+   * @param dryRunEnabled Dry Run boolean flag
+   * @return Pageable to pass to query. In the case of dry runs, this is the next page.
+   * @implNote This parent handler expects the child's cleanupMetadata call to update & remove the record from this call
+   * such that subsequent DB queries will not return the record. Hence why we only call next during dryRuns
+   * where no updates occur.
+   * @implNote Note that we only expect pageable.next to be called during a dry run.
+   */
+  public Pageable processPage(Pageable pageable, LocalDateTime instant, Page<HousekeepingMetadata> page,
+      boolean dryRunEnabled) {
+    List<HousekeepingMetadata> pageContent = page.getContent();
+    if (dryRunEnabled) {
+      pageContent.forEach(metadata -> metadataCleanup.cleanupMetadata(metadata, instant, true));
```

Comment:
this is weird with the two methods, also instead of true just pass along the dryRunEnable flag.